### PR TITLE
Enhance/7

### LINKED ISSUE #7: Enhance get game function
Add more flexibility to the get game function, so decks that might not be 100% balanced (but playable) can be obtained from the function.

### PR DESCRIPTION
**Pull request by vetom**
This pull request closes #7. When we are crafting a new game, the function will return the game with the maximum number of players found during the process. The balance will between -9 and 9 when not all the players are playing.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,15 +21,21 @@ exports.getGameFromTemplate = function (players, template) {
...
+        return templateCards.indexOf(card.role) >= 0;
     });
 }
 
 function _getBalancedGame(players, chosenCards) {
     chosenCards = _classifyCards(chosenCards || cards.all);
-    while (currentWeight < -1 || currentWeight > 1) {
+    var flex = 1;
+    var tries = 0;
+    while (currentWeight < -1 * flex || currentWeight > 1 * flex) {
```
Comment (wistcc) [inline index.js]: Remove `1*flex` and just leave `flex`

File: index.js [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
 var templates = require('./templates');
 var availableCards = {};
-var deck = {};
-var currentWeight = 100;
+var game = {
+    deck : {},
+    weight : 100,
+    players: 0
+};
+var candidate = {
```
Comment (wistcc) [inline index.js]: Should be called `gameCandite`

**Review by wistcc — CHANGES_REQUESTED** [review]

File: index.js [hunk 3]
```diff
@@ -21,21 +30,24 @@ exports.getGameFromTemplate = function (players, template) {
...
+        return templateCards.indexOf(card.role) >= 0;
     });
 }
 
 function _getBalancedGame(players, chosenCards) {
     chosenCards = _classifyCards(chosenCards || cards.all);
-    while (currentWeight < -1 || currentWeight > 1) {
+    var flex = 1;
+    var tries = 0;
+    while (!allPlayers || game.weight < -1 * flex || game.weight > 1 * flex) {
```
Comment (wistcc) [inline index.js]: Remove `1*flex` and just leave `flex`

**Review by wistcc — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (vetom) [reply index.js]: `gameCandidate`

**Review by wistcc — APPROVED** [review]

**Pull request merged**